File: Cpp_chunks/sudoku_cell.cpp
class SudokuCell {
public:
    int value;
    Set *avSet;
    bool fixed;

    bool hasValue(){
        return value != 0;
    }
    void safeAvSetRemove(int candidate){
        if(!hasValue()  && avSet->has(candidate)){
            this->avSet->remove(candidate);
        }
    }
};

File: Python/chunks/chapter_2/introduction/sudoku_cell.py

class SudokuCell:
    n = 9
    def __init__(self, value=0, fixed=False):
        self.value = value
        self.fixed = fixed
        self.av_set = None
        if not fixed:
            self.av_set = set()

    def has_value(self):
        return self.value != 0

    def av_set_remove(self, value):
        if not self.has_value() and (value in self.av_set):
            self.av_set.remove(value)
